components/director/internal/open_resource_discovery/service.go

```diff
@@ -380,57 +400,88 @@ func (s *Service) processDocuments(ctx context.Context, app *model.Application,
 	}
 
 	ordLocalID := s.getUniqueLocalTenantID(documents)
-	if ordLocalID != "" && app.LocalTenantID == nil {
-		if err := s.appSvc.Update(ctx, app.ID, model.ApplicationUpdateInput{LocalTenantID: str.Ptr(ordLocalID)}); err != nil {
+	if ordLocalID != "" && resource.LocalTenantID == nil {
+		if err := s.appSvc.Update(ctx, resource.ID, model.ApplicationUpdateInput{LocalTenantID: str.Ptr(ordLocalID)}); err != nil {
 			return err
 		}
 	}
+	for _, doc := range documents {
+		if resource.Type == directorresource.ApplicationTemplate && doc.DescribedSystemVersion == nil {
+			continue
+		}
 
-	vendorsFromDB, err := s.processVendors(ctx, app.ID, vendorsInput)
-	if err != nil {
-		return err
-	}
+		resourceToAggregate := Resource{
+			ID:   resource.ID,
+			Type: directorresource.Application,
+		}
 
-	productsFromDB, err := s.processProducts(ctx, app.ID, productsInput)
-	if err != nil {
-		return err
-	}
+		if doc.DescribedSystemVersion != nil {
+			applicationTemplateID := resource.ID
+			if resource.Type == directorresource.Application && resource.ParentID != nil {
```

Comment:
what if `resource.Type == directorresource.Application`, but `resource.ParentID == nil` then on l424 i will try to get an `applicationTemplateID` will be equal to `resource.ID` (l419), but `resource.Type` is `Application` so the function will be called with wrong value for `applciationTemplateID`

Reply:
In this case the aggregation will fail because there will be no Application Template with ID `resource.ID` (which is the App's ID). And that probably is fine, because the document expects to be aggregated for static data (doc.DescribedSystemVersion) but we do not know for whom (for which App Template) the aggregation is happening.